sql/rls: address post-merge review comments from PR #146292

spilchen opened the pull request:
This change applies a set of late review suggestions from Michael regarding the previous RLS-related changes merged in PR #146292.

Epic: CRDB-11724
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146393)
<!-- Reviewable:end -->

michae2 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thank you!

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146393#-:-OPm1_DD3mE8MgvdW-cG:b-ae66e1)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen committed:
sql/rls: address post-merge review comments from PR #146292

This change applies a set of late review suggestions from Michael
regarding the previous RLS-related changes merged in PR #146292.

Epic: CRDB-11724
Release note: none

spilchen commented:
TFTR!

bors r+
